packages/pluginutils/types/index.d.ts

```diff
@@ -0,0 +1,49 @@
+import { Node } from 'estree-walker';
+
+export interface AttachedScope {
+  parent?: AttachedScope;
+  isBlockScope: boolean;
+  declarations: { [key: string]: boolean };
+  addDeclaration(node: Node, isBlockDeclaration: boolean, isVar: boolean): void;
+  contains(name: string): boolean;
+}
+
+export interface DataToEsmOptions {
+  compact?: boolean;
+  indent?: string;
+  namedExports?: boolean;
+  objectShorthand?: boolean;
+  preferConst?: boolean;
+}
+
+export type AddExtension = (filename: string, ext?: string) => string;
+export const addExtension: AddExtension;
+
+export type AttachScopes = (ast: Node, propertyName?: string) => AttachedScope;
+export const attachScopes: AttachScopes;
+
+export type CreateFilter = (
+  include?: Array<string | RegExp> | string | RegExp | null,
```

Comment:
`Array<string | RegExp> | string | RegExp | null` is used by other packages with `include` and `exclude` filters, so maybe it should be exported as a helper type from pluginutils.

```ts
export const Filter = Array<string | RegExp> | string | RegExp | null;
```

Reply:
I was thinking about that this morning actually. We've got some work to do to improve this module in the future. Right now I'd like to keep the focus on getting the plugin migrated as-is., since it's been functional.